src/nvim/getchar.c

```diff
@@ -2080,7 +2080,7 @@ static int vgetorpeek(const int advance)
             xfree(save_m_keys);
             xfree(save_m_str);
             if (i == FAIL) {
-              break;
+              continue;
```

Comment:
The thing is that `break` breaks out of the `for(;;)` loop, and into the `while()` loop. Because of the current state in the function this immediately goes back into the `for(;;)` loop and does no different from a `continue` (except in the special case mentioned in the commit message).

Having two different lines implementing the same functionality seems unnecessary.